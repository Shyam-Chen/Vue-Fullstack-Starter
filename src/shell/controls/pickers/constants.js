// @flow

export const INITIAL = {
  date: null,
  dateMenu: false,
  dateDialog: false,

  time: null,
  timeMenu: false,
  timeDialog: false,
};
