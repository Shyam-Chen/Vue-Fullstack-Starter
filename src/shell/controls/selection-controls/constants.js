export const INITIAL = {
  // checkboxes
  technologies: [],
  technologiesList: ['ECMAScript', 'HTML5', 'Node.js', 'Docker'],

  // radios
  gender: '',
  genderList: ['Male', 'Female', ' Other'],

  // switch
  autoplay: false,

  // slider
  volume: 0,
};
