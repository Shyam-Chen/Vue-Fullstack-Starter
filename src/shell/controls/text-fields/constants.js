// @flow

import { email } from '~/shared/utils/regexp';

export const INITIAL = {
  nickname: '',
  nicknameRules: [
    () => !!INITIAL.nickname || 'This field is required',
    () => INITIAL.nickname.length <= 15 || 'Word length is too long',
  ],
  creditCard: '',
  phone: '',
  email: '',
  emailRules: [
    () => !!INITIAL.email || 'This field is required',
    () => email.test(INITIAL.email) || 'Email format error',
  ],
};
