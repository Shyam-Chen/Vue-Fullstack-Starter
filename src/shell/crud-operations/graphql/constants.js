export const INITIAL = {
  dataset: [],
  searchData: {},
};
