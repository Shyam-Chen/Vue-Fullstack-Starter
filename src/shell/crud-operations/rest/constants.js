export const API_LIST = `${process.env.API_URL}/text-list`;

export const INITIAL = {
  selected: [],
  dataset: [],
  addData: { text: '' },
  searchData: { text: '' },
  editData: { id: 0, text: '', dialog: false },
  deleteData: { id: 0, dialog: false },
  loading: false,
};
