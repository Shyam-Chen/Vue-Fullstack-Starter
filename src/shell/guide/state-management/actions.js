// @flow

import { ActionContext } from 'vuex';

import { ICounter } from './types';

export default {
  // sync action
  increment({ state, commit }: ActionContext<ICounter>): void {
    commit('increment', state.step);
  },
  decrement({ state, commit }: ActionContext<ICounter>): void {
    commit('decrement', state.step);
  },
  // thunk action
  incrementAsync({ dispatch }: ActionContext<ICounter>): void {
    setTimeout(() => dispatch('increment'), 1000);
  },
  decrementAsync({ dispatch }: ActionContext<ICounter>): void {
    setTimeout(() => dispatch('decrement'), 1000);
  },
  incrementIfOdd({ state, dispatch }: ActionContext<ICounter>): void {
    if (Math.abs(state.value % 2) === 1) {
      dispatch('increment');
    }
  },
  // observable action
  decrementIfEven({ state, dispatch }: ActionContext<ICounter>): void {
    if (Math.abs(state.value % 2) === 0) {
      dispatch('decrement');
    }
  },
};
