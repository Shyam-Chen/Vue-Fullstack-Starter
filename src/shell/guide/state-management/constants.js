// @flow

import { ICounter } from './types';

export const INITIAL: ICounter = {
  value: 0,
  step: 1,
  stepList: [...Array(10).fill(1).map((x, y) => x + y)],
};
