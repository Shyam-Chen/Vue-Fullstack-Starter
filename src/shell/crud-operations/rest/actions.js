import axios from 'axios';

import { API_LIST } from './constants';

export default {
  success({ commit }, data) {
    commit('success', data);
  },
  failure({ commit }, error) {
    commit('failure', error);
  },

  addItem({ dispatch, state }, text) {
    if (!text) return;

    state.loading = true;

    axios
      .post(API_LIST, { text })
      .then(() => {
        state.addData.text = '';
        dispatch('searchItem');
      })
      .catch(error => dispatch('failure', error));
  },
  searchItem({ dispatch, state }, text) {
    state.loading = true;

    axios
      .get(API_LIST, { params: { text } })
      .then(res => {
        state.loading = false;
        state.searchData.text = '';
        dispatch('success', res.data);
      })
      .catch(error => dispatch('failure', error));
  },
  editItem({ dispatch, state }, item) {
    if (item.text) {
      state.editData.dialog = false;
      state.loading = true;

      axios
        .put(`${API_LIST}/${item.id}`, { text: item.text })
        .then(() => dispatch('searchItem'))
        .catch(error => dispatch('failure', error));
    }
  },
  deleteItem({ dispatch, state }, id) {
    state.deleteData.dialog = false;
    state.loading = true;

    axios
      .delete(`${API_LIST}/${id}`)
      .then(() => dispatch('searchItem'))
      .catch(error => dispatch('failure', error));
  },
};
