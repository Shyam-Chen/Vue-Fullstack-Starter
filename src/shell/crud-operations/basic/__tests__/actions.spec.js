import { INITIAL } from '../constants';
import actions from '../actions';

describe('actions', () => {
  it('should handle addItem', () => {
    const state = { ...INITIAL };

    const payload = {
      primary: 'foo',
      accent: 'bar',
    };

    actions.addItem({ state }, payload);

    expect(state.dataset.length).toBe(5);
    expect(state.dataset).toMatchSnapshot();
  });

  it('should handle editItem', () => {
    const state = { ...INITIAL };

    const payload = {
      id: 1,
      primary: 'foo',
      accent: 'bar',
    };

    actions.editItem({ state }, payload);

    expect(state.dataset.length).toBe(4);
    expect(state.dataset).toMatchSnapshot();
  });

  it('should handle deleteItem', () => {
    const state = { ...INITIAL };
    const payload = { id: 1 };

    actions.deleteItem({ state }, payload);

    expect(state.dataset.length).toBe(3);
    expect(state.dataset).toMatchSnapshot();
  });

  it('should handle deleteChecked', () => {
    const state = { ...INITIAL };
    const selected = [...INITIAL.dataset];

    actions.deleteChecked({ state }, selected);

    expect(state.dataset.length).toBe(0);
  });

  it('should handle dialog', () => {
    const state = { ...INITIAL };
    const data = { dialogs: { edit: true } };

    actions.setState({ state }, data);

    expect(state.dialogs.edit).toBe(true);
  });

  it('should handle setState', () => {
    const state = { ...INITIAL };
    const item = { id: 1, primary: 'foo', accent: 'bar' };
    const data = { editData: item };

    actions.setState({ state }, data);

    expect(state.editData).toEqual(item);
  });
});
