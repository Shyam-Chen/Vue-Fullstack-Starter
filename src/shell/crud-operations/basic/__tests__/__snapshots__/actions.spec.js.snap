// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should handle addItem 1`] = `
Array [
  Object {
    "accent": "MobX",
    "id": 1,
    "primary": "Vanilla",
  },
  Object {
    "accent": "NGXS",
    "id": 2,
    "primary": "Angular",
  },
  Object {
    "accent": "Redux",
    "id": 3,
    "primary": "React",
  },
  Object {
    "accent": "Vuex",
    "id": 4,
    "primary": "Vue",
  },
  Object {
    "accent": "bar",
    "id": 5,
    "primary": "foo",
  },
]
`;

exports[`actions should handle deleteItem 1`] = `
Array [
  Object {
    "accent": "NGXS",
    "id": 2,
    "primary": "Angular",
  },
  Object {
    "accent": "Redux",
    "id": 3,
    "primary": "React",
  },
  Object {
    "accent": "Vuex",
    "id": 4,
    "primary": "Vue",
  },
]
`;

exports[`actions should handle editItem 1`] = `
Array [
  Object {
    "accent": "bar",
    "id": 1,
    "primary": "foo",
  },
  Object {
    "accent": "NGXS",
    "id": 2,
    "primary": "Angular",
  },
  Object {
    "accent": "Redux",
    "id": 3,
    "primary": "React",
  },
  Object {
    "accent": "Vuex",
    "id": 4,
    "primary": "Vue",
  },
]
`;
